### PR TITLE
Improvement:command-refactor

### PR DESCRIPTION
**Pull request by siddhant-basepair**
This PR will change commands -

Updated Commands will be - 

Module
module get
module delete
module create
module update
module list
Can pass `-p or –pipeline` flag to list modules for specific pipeline

Pipeline
pipeline get
pipeline delete
pipeline create
pipeline update
pipeline list

Genome
genome get
genome list
                  
Sample
sample get
sample delete
sample create
sample update
sample download
sample list
Can pass `--project-id` flag to filter samples for a specific project

Project
project create
project update

File
file download

Analysis
analysis get
analysis delete
analysis create
analysis update
analysis download 
analysis download log
analysis reanalyze
analysis list

Related Ticket -
https://app.asana.com/0/1201368373933132/1201611319941112

### CODE REVIEW

**Review by Lalee10 — CHANGES_REQUESTED** [review]
- Many of the help messages are not accurate. Help messages for the flag `-u, --uid` are not accurate for many commands like module delete, module pipeline, file download etc. Please fix them and any other help messages.
- Run the help messages of each command at least once to verify your changes

File: bin/basepair [hunk 1]
```diff
@@ -630,280 +711,242 @@ def read_args():
...
-
-  # download parsers
-  download_p = actions_p.add_parser('download', help='Download items.')
-
-  download_sp = download_p.add_subparsers(
-    help='Type of item to download',
-    dest='download_type'
+  list_analyses_p.add_argument(
+    '-p',
+    '--project',
```
[the remark this answers is not in the document]
Comment (siddhant-basepair) [reply bin/basepair:739]: it was earlier like this so please tell me if this also needs to  be refactored.

[the remark this answers is not in the document]
Comment (Lalee10) [reply bin/basepair:739]: Yes, it does need to be refactored

[the remark this answers is not in the document]
Comment (Lalee10) [reply bin/basepair:739]: @siddhant-basepair this requirement has changed again, please define the args as per the updated doc

**Review by blasvicco — CHANGES_REQUESTED** [review]
I believe the file will become more readable if we move the argument parser to a separated file or module.

File: bin/basepair [hunk 2]
```diff
@@ -95,16 +95,34 @@ def main():
     verbose=args.verbose
   )
 
-  if args.action == 'create':
-    if args.create_type == 'project':
-      create_project(bp, args)
-    if args.create_type == 'module':
+  if args.datatype == 'module':
```
[the remark this answers is not in the document]
Comment (siddhant-basepair) [reply bin/basepair:98]: it really helped to refactor the CLI to an optimal state now it's much easier to work on.

**Pull request closed without merging**